### ISSUE TITLE
Command line output is unreadable with white background

### ISSUE THREAD

**Issue by paulpach** [dotnet/docfx#5309]
I have mild astigmatism,  for that reason, I prefer to have my terminal with [white background](https://jessicaotis.com/academia/never-use-white-text-on-a-black-background-astygmatism-and-conference-slides/).  docfx does not work property with white background.

**Operation System**: MacOS

**DocFX Version Used**: 2.47.0.0

**Template used**: any

**Steps to Reproduce**:

1. Configure your terminal for white background
2. run `docfx --version` or `docfx --help`
3. docfx display the version in black font,  perfectly fine
4. run `docfx init`  or `docfx` in an existing project
5. font is white, which is unreadable on white background.

**Expected Behavior**:
docfx should use black font when the terminal is set to white background.
In other words docfx should use the default font color for the terminal

**Actual Behavior**:

Text is white or light yellow for warnings.   Completely unreadable. See screenshots
<img width="1250" alt="Screen Shot 2019-11-20 at 9 03 21 AM" src="https://user-images.githubusercontent.com/466007/69250858-108ddb00-0b76-11ea-94c4-4bf760f04e2a.png">
<img width="1250" alt="Screen Shot 2019-11-20 at 9 13 28 AM" src="https://user-images.githubusercontent.com/466007/69250866-15eb2580-0b76-11ea-954c-f41bde2383b8.png">

<img width="1250" alt="Screen Shot 2019-11-20 at 9 15 29 AM" src="https://user-images.githubusercontent.com/466007/69251006-53e84980-0b76-11ea-96b9-d61d98fb870e.png">

**Comment by KalleOlaviNiemitalo** [comment]
Using the default text color instead of white would be relatively easy, I think. Search for assignments to `Console.ForegroundColor` and make them recognize `ConsoleColor.White` and instead call `Console.ResetColor()`.

Avoiding yellow text while keeping the default white background seems much harder. The Mono implementation apparently [assumes that the default colors are white on black](https://github.com/mono/mono/blob/73ae9dbc3693201939efce02b2b36c4ce5b352bd/mcs/class/corlib/System/TermInfoDriver.cs#L90-L91) and does not attempt to query these defaults from the terminal itself or from the terminfo database; I don't think they even support such a query. So DocFX cannot just check `Console.BackgroundColor` on startup and adjust its use of colors accordingly.

I think the easiest workaround for you would be to set `TERM=dumb` in the environment when you run DocFX; then Mono should ignore DocFX's attempts to change the colors.

**Comment by KalleOlaviNiemitalo** [comment]
In fact, xterm documents a query for the background color, under [Operating System Commands](https://invisible-island.net/xterm/ctlseqs/ctlseqs.html#h3-Operating-System-Commands). I tested that mintty 3.1.0 (x86_64-pc-msys) also supports the following:

* Output to the terminal, in C syntax: `"\033]11;?\033\\"`
* Sample response from the terminal, in C syntax: `"\033]11;rgb:0000/0000/0000\033\\"`

In principle, DocFX could send this query, wait for a response, parse the color, and check whether it is dark or light. There are a few problems, though:

* The format of the color is not very strictly specified. A terminal could conceivably respond with a named color or with coordinates in some color space other than "rgb". In practice, making DocFX support a few popular formats should suffice.
* If the terminal does not support this query, then it will not respond at all, so DocFX should not wait forever for a response. As a workaround, DocFX could send first this color query and then something more widely supported, like the Device Attributes query `"\033[0c"`. If DocFX got the Device Attributes response first, then it would know not to wait for the color response. However, DocFX would then have to read the entire Device Attributes response.
* If the user starts DocFX and presses a key before DocFX sends the color query, then DocFX will receive the keypress before the color response. This could make DocFX believe there will be no color response. After DocFX exits without reading the color response, the user's shell would then receive the color response and possibly treat it as part of a command. I don't know any good way to fix that.
* If DocFX is being run on a build server and the output has been redirected to a file, then no terminal will ever respond. DocFX could perhaps recognize this by checking what type of device its standard output refers to, plus have a timeout just in case.

It all seems quite complex to implement. It would be much easier to make DocFX recognize some environment variable, e.g. `DOCFX_TERMINAL_BACKGROUND=light`. But then again, I think `TERM=dumb` is already available.

**Comment by paulpach** [comment]
Thank you for looking into it.  The workaround is certainly better than nothing.  

Seems finding out the color is a bit of a hack,  Ideally,  mono should do the query.   Does .Net core have the same issue?   I have not checked.

**Comment by KalleOlaviNiemitalo** [comment]
See <https://github.com/dotnet/corefx/commit/14f36001ee138522f496318fc685d37c9f3ab9ae> for .NET Core. It does not try to query the terminal for the colors on Unix; instead, the `Console.BackgroundColor` and `Console.ForegroundColor` properties return `(ConsoleColor)-1` until the application sets them.

The difficulty in querying the terminal for colors or anything else is caused by the standard ECMA-48 terminal protocol and its variants, not by any software in the computer that is communicating with the terminal. I don't think Mono or .NET Core should automatically send any queries to the terminal; that would cause the same problems as having DocFX itself send them, and other applications would also suffer. Queries seem more reasonable to use in a program that reads keypresses from the terminal as part of its normal operation, like a text editor.

**Comment by yufeih** [comment]
For ordinary output text, leaving console output color as default should suffice. 

For other text that needs color, querying terminal background is none-trivial. I would suggest picking a color that stands out in both black and white backgrounds. `DarkYellow` is one of them.

**Comment by KalleOlaviNiemitalo** [comment]
Perhaps DocFX could emit SGR control sequences like `"\033[1m"` (bold) or `"\033[4m"` (underlined) instead of choosing specific colors, at least on macOS and Linux where terminals usually support these.

**Comment by yufeih** [comment]
I played with NodeJs a little bit on Mac, and the text color change as I'm changing screen background, including text that are already printed. Feels magical to be. @KalleOlaviNiemitalo , do you happen to know that technique they are using?

**Comment by KalleOlaviNiemitalo** [comment]
@yufeih, I don't know how NodeJs does it, but I assume it uses bold etc. like in my previous comment.

**Comment by paulpach** [comment]
I can confirm bold and underline work fine in mac os x with iterm and iterm2 in all color schemes (not just white and black)
